docs/victoriametrics/relabeling.md

```diff
@@ -10,250 +10,383 @@ aliases:
+
+For example, you may want to scrape only the targets with the label `env=prod`:
+
+```yaml {hl_lines="2"}
+relabel_configs:
+- source_labels: [env]
+  regex: prod
+  action: keep
 ```
 
-See also:
+This keeps only targets where the env label is `prod`, and drops the rest.
+
+**2. Scraping Relabeling**
 
-- [how to remove labels from scraped metrics](#how-to-remove-labels-from-scraped-metrics)
-- [useful tips for metric relabeling](#useful-tips-for-metric-relabeling)
+Once VictoriaMetrics has finished selecting the targets using `relabel_configs`, it starts scraping those endpoints. 
 
+After scraping, you can apply `metric_relabel_configs`. This is the second stage, and it operates on **individual metrics** that were just scraped from the targets, not the targets themselves. This means you can filter or modify the scraped time series before VictoriaMetrics stores them in its time series database.
 
-## How to rename scraped metrics
+**3. Remote Write Relabeling**
 
-Metric name is a regular label with special name - `__name__` (see [these docs](https://docs.victoriametrics.com/keyconcepts/#labels)).
-So renaming of metric name is performed in the same way as changing label value.
+This happens after `metric_relabel_configs` have been processed, just before the metrics are sent to a specific `remoteWrite.url` destination configured in vmagent.
 
-Let's look at a few examples.
+The purpose of this stage is to apply destination-specific relabeling rules. This is a key feature for routing and filtering data differently for multiple backends. For example, you could use this stage to:
 
-The following config renames `foo` metric to `bar` across all the scraped metrics, while leaving other metric names as is:
+- Send only metrics with `env=prod` to your production VictoriaMetrics cluster.
+- Send only metrics with `env=dev` to a development cluster.
+- Send a subset of metrics (e.g., only specific high-importance ones) to a Kafka topic for real-time processing, while sending all metrics to long-term storage.
+- Drop certain labels only for metrics going to one specific backend but keep them for another.
+
+## Relabling Actions
+
+### Relabeling Enhancements
+
+`vmagent` provides the following enhancements on top of Prometheus-compatible relabeling:
+
+* The `replacement` field allows constructing new label values by referencing existing ones using the `{{label_name}}` syntax. For example, if a metric has the labels `{instance="host123", job="node_exporter"}`, this rule will set the `instance-job` label to `host123-node_exporter` ([Try it](https://play.victoriametrics.com/select/0/prometheus/graph/#/relabeling?config=-+target_label%3A+%22instance-job%22%0A++replacement%3A+%22%7B%7Binstance%7D%7D-%7B%7Bjob%7D%7D%22&labels=node_cpu_seconds_total%7Bcpu%3D%220%22%2C+instance%3D%22server-1%3A9100%22%2C+job%3D%22node_exporter%22%2C+mode%3D%22idle%22%7D)):
```

Comment:
@func25 will have to change the playground links to respect https://github.com/VictoriaMetrics/VictoriaMetrics/pull/8770 change
This change forbids specifying metric name out of `{}` braces. So 
```
node_cpu_seconds_total{cpu="0", instance="server-1:9100", job="node_exporter", mode="idle"}
```
should become:
```
{__name__="node_cpu_seconds_total", cpu="0", instance="server-1:9100", job="node_exporter", mode="idle"}
```